core/net/mac/framer-802154.c

```diff
@@ -138,43 +141,46 @@ create_frame(int type, int do_create)
   }
 
   /* Complete the addressing fields. */
-  /**
-     \todo For phase 1 the addresses are all long. We'll need a mechanism
-     in the rime attributes to tell the mac to use long or short for phase 2.
-   */
-  if(LINKADDR_SIZE == 2) {
-    /* Use short address mode if linkaddr size is short. */
-    params.fcf.src_addr_mode = FRAME802154_SHORTADDRMODE;
+  if((params.fcf.frame_type == FRAME802154_BEACONFRAME ||
+      params.fcf.frame_type == FRAME802154_ACKFRAME) &&
+     packetbuf_attr(PACKETBUF_ATTR_MAC_NO_SRC_ADDR) == 1) {
```

Comment:
Ah, that's a good point. Since I was rewriting tsch_packet_create_eack() and tsch_packet_create_eb() with this change, I enabled it for beacon and ack for safety. But, seeing the code now, I think these conditions can be removed.